### PR TITLE
Refactor osm code

### PR DESCRIPTION
**Pull request by dkniffin**

### CODE REVIEW

File: app/models/concerns/normalize.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# Normalizes various similar datatypes
+module Normalize
+  include Geographic
+
+  @@factory = RGeo::Geographic.spherical_factory(srid: 4326)
```
Comment (dkniffin) [inline app/models/concerns/normalize.rb]: We only ever need to define this once, so this is what I want.

File: app/models/concerns/geographic/polygon.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        # Set SRID
+        point = Geographic.set_srid(rgeo_point)
+        geometry = Geographic.set_srid(geometry_column)
+
+        # Do the Query
+        where(geometry.st_contains(point))
+      }
+
+      # Returns the objects that intersect with the given line
+      scope :intersected_by_line, -> (arg1, arg2 = nil) {
```
Comment (dkniffin) [inline app/models/concerns/geographic/polygon.rb]: I think I need to update hound configs to match SB's style

File: spec/models/concerns/geographic/polygon.rb [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      let(:geometry3) { Normalize.to_rgeo_polygon(geometry3_points) }
+      let!(:object3) do
+        FactoryGirl.create(options[:factory], options[:geometry_attribute] => geometry3)
+      end
+      let(:arg1) { target_point_array }
+
+      it 'returns the ones that contain the point' do
+        expect(subject.ids).to include(object.id)
+        expect(subject.ids).to include(object2.id)
+        expect(subject.ids).to_not include(object3.id)
```
Comment (prawl) [inline spec/models/concerns/geographic/polygon.rb]: I would use `expect(subject.ids).to contain_exactly(object.id, object2.id)`

Comment (dkniffin) [inline spec/models/concerns/geographic/polygon.rb]: :+1:

**Comment by prawl** [comment]
This looks good.  Looking into the failing tests.

**Pull request merged**